ngboost/ngboost.py

```diff
@@ -395,9 +395,24 @@ def feature_importances_(self):
             return None
         # Reshape the base_models
         params_trees = zip(*self.base_models)
+
         # Get the feature_importances_ for all the params and all the trees
+
+        def _get_feature_importance(tree, tree_index):
+            tree_feature_importance = getattr(tree, "feature_importances_")
+            if self.col_sample == 1.:
+                return tree_feature_importance
+
+            n_features = np.max(self.col_idxs) + 1
```

Comment:
instead of doing this, let's add a line at the beginning of `NGBoost.fit()` that sets `self.n_features = X.shape[1]` and then we can reference `self.n_features` instead of computing it in this function.

Reply:
Didn't want to touch other functions in the beginning, but doing `np.max` this way is less clear and could given unintended results for some corner cases i think. Yeah setting it in `fit` is better.